services/vision/mlvision/detector.go

```diff
@@ -47,42 +49,42 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 	return func(ctx context.Context, img image.Image) ([]objectdetection.Detection, error) {
 		origW, origH := img.Bounds().Dx(), img.Bounds().Dy()
 		resized := resize.Resize(inWidth, inHeight, img, resize.Bilinear)
-		inMap := make(map[string]interface{})
+		inMap := ml.Tensors{}
 		switch inType {
 		case UInt8:
-			inMap["image"] = rimage.ImageToUInt8Buffer(resized)
+			inMap["image"] = tensor.New(
+				tensor.WithShape(1, int(inHeight), int(inWidth), 3),
```

Comment:
This leading `1` here is acting as the batching dimension, so having this hardcoded constrains the detector/classifier to models that accept that leading batching dimension. Just noting it. It works for what I need for now, but looks like it forms part of the implicit interface between the vision services and mlmodels.

Reply:
so many models have the n_batch input dimension because of the training step they go through -- It's almost always present. But yes, it definitely should be noted when creating the tutorial for people who want to build models that can be used with the vision service